### PR TITLE
[LIGHTSPD-34] fixed a bug where overnight sync was syncing with incorrect values

### PR DESCRIPTION
**Pull request by whc105**
-Resolves the daily sync quantity misalignment issue
-Added splunk logging
-Added sync reset

How to verify
-Find an item that has different quantities in lightspeed and vtex
-Wait (or run the process-sync)
-Verify that the item quantity has been updated correctly on vtex

### CODE REVIEW

File: node/utils/SyncCatalog.ts [hunk 1]
```diff
@@ -351,6 +361,11 @@ export async function updatePriceAndInventory({
     inventoryResult = values
   })
 
+  logger.info({
+    data: updatedItemsInventory,
+    message: 'SyncCatalog-updatedItemsInventory',
+  })
+
```
Comment (arturmagalhaesjr) [inline node/utils/SyncCatalog.ts:368]: There is a logger.info that is logging the updated items inventory, however, there are many promises which are no catch or try statements. I'd catch them and put the logger.error at least.

File: node/utils/SyncCatalog.ts [hunk 2]
```diff
@@ -302,6 +303,11 @@ export async function updatePriceAndInventory({
     itemQueryResult = values
   })
 
+  const skuQuantityRemap: any = {}
+  itemQueryResult.forEach((item: any) => {
+    skuQuantityRemap[item.sku] = item.newQuantity
```
Comment (arturmagalhaesjr) [inline node/utils/SyncCatalog.ts:308]:
``` suggestion
    if (item?.sku) {
       skuQuantityRemap[item.sku] = item.newQuantity
    }
```

**Review by arturmagalhaesjr — CHANGES_REQUESTED** [review]

File: node/utils/SyncCatalog.ts (hunk 2 above)
Comment (whc105) [reply node/utils/SyncCatalog.ts:308]: I don't think we need to check the SKU. The sku field should always be present in the response
https://developers.vtex.com/vtex-rest-api/reference/inventorybysku

File: node/utils/SyncCatalog.ts (hunk 1 above)
Comment (whc105) [reply node/utils/SyncCatalog.ts]: Resolved in new commit

**Review by wender — APPROVED** [review]

**Pull request merged**